File: apps/backend-go/internal/model/datasource.go
package model

import (
	"encoding/json"
	"time"
)

// Datasource mirrors the datasources table.
//
// Config uses json.RawMessage (not []byte) so encoding/json embeds it as a nested
// object — []byte would be base64-encoded by the default marshaler and break the
// BFF/engine contract (same lesson as model.Page.Schema; see page_test.go Bug1
// regression). Aligned with Java entity.Datasource (configJson as String) and the
// shared §9.3 DDL.
type Datasource struct {
	ID        string          `db:"id" json:"id"`
	SiteID    string          `db:"site_id" json:"siteId"`
	Name      string          `db:"name" json:"name"`
	Type      string          `db:"type" json:"type"`
	Config    json.RawMessage `db:"config_json" json:"config"`
	CreatedAt time.Time       `db:"created_at" json:"createdAt"`
	UpdatedAt time.Time       `db:"updated_at" json:"updatedAt"`
}

File: apps/backend-java/src/main/java/com/luban/backend/entity/Datasource.java
package com.luban.backend.entity;

import java.time.Instant;

/**
 * Datasource entity; table datasources. configJson holds JSON string.
 *
 * <p>Aligned with luban-backend-go internal/model/datasource.go (Config json.RawMessage).
 * {@code type} whitelist ({@code static|api}) is enforced in the service layer, not here.
 */
public class Datasource {
    private String id;
    private String siteId;
    private String name;
    private String type;
    private String configJson;
    private Instant createdAt;
    private Instant updatedAt;

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getSiteId() { return siteId; }
    public void setSiteId(String siteId) { this.siteId = siteId; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public String getConfigJson() { return configJson; }
    public void setConfigJson(String configJson) { this.configJson = configJson; }
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
}
